include/util/static_rtree.hpp

```diff
@@ -158,189 +127,329 @@ class StaticRTree
+                // std::clog << "M for the root node is " << M << std::endl;
+            }
+            else
+            {
+                // std::clog << "processing a tree node with parent " << r.parent << std::endl;
+                BOOST_ASSERT(m_search_tree[r.parent].child_count < BRANCHING_FACTOR);
+                if (m_search_tree[r.parent].child_count == 0)
                 {
-                    const std::uint32_t input_object_index =
-                        input_wrapper_vector[wrapped_element_index].m_array_index;
-                    const EdgeDataT &object = input_data_vector[input_object_index];
-
-                    current_leaf.object_count += 1;
-                    current_leaf.objects[object_index] = object;
-
-                    Coordinate projected_u{
-                        web_mercator::fromWGS84(Coordinate{m_coordinate_list[object.u]})};
-                    Coordinate projected_v{
-                        web_mercator::fromWGS84(Coordinate{m_coordinate_list[object.v]})};
-
-                    BOOST_ASSERT(std::abs(toFloating(projected_u.lon).operator double()) <= 180.);
-                    BOOST_ASSERT(std::abs(toFloating(projected_u.lat).operator double()) <= 180.);
-                    BOOST_ASSERT(std::abs(toFloating(projected_v.lon).operator double()) <= 180.);
-                    BOOST_ASSERT(std::abs(toFloating(projected_v.lat).operator double()) <= 180.);
-
-                    rectangle.min_lon =
-                        std::min(rectangle.min_lon, std::min(projected_u.lon, projected_v.lon));
-                    rectangle.max_lon =
-                        std::max(rectangle.max_lon, std::max(projected_u.lon, projected_v.lon));
-
-                    rectangle.min_lat =
-                        std::min(rectangle.min_lat, std::min(projected_u.lat, projected_v.lat));
-                    rectangle.max_lat =
-                        std::max(rectangle.max_lat, std::max(projected_u.lat, projected_v.lat));
-
-                    BOOST_ASSERT(rectangle.IsValid());
+                    /*
+                    std::clog << r.parent << " has child count of 0, setting to "
+                              << m_search_tree.size() - 1 << std::endl;
+                              */
+                    m_search_tree[r.parent].first_child_index = m_search_tree.size() - 1;
                 }
+                ++m_search_tree[r.parent].child_count;
+            }
 
-                // append the leaf node to the current tree node
-                current_node.child_count += 1;
-                current_node.children[leaf_index] =
-                    TreeIndex{node_index * BRANCHING_FACTOR + leaf_index, true};
-                current_node.minimum_bounding_rectangle.MergeBoundingBoxes(
-                    current_leaf.minimum_bounding_rectangle);
+            std::size_t N2 = std::ceil(static_cast<double>(N) / M);
+            std::size_t N1 = N2 * std::ceil(std::sqrt(M));
```

Comment:
`N1` and `N2` make the following code rather hard to follow. `N2` is the number of children, right?